Docs: Improve v8 upgrade notes for SQL data sources (#38792)
grafana/grafana: docs/sources/datasources/mssql.md

## Time series queries

> If you set Format as to _Time series_, then the query must have a column named time that returns either a SQL datetime or any numeric datatype representing Unix epoch in seconds. In addition, resultsets of time series queries must be sorted by time for panels to properly visualize the result.

Reviewer: ```suggestion
If you set Format as to _Time series_, then the query must have a column named time that returns either a SQL datetime or any numeric datatype representing Unix epoch in seconds. In addition, result sets of time series queries must be sorted by time for panels to properly visualize the result.
```

> A time series query result is returned in a [wide data frame format]({{< relref "../developers/plugins/data-frames.md#wide-format" >}}). Any column except time and those of type string is transformed to value fields in the data frame query result. Any string column is transformed into field labels in the data frame query result.

Reviewer: ```suggestion
A time series query result is returned in a [wide data frame format]({{< relref "../developers/plugins/data-frames.md#wide-format" >}}). Any column except time or of type string transforms into value fields in the data frame query result. Any string column transforms into field labels in the data frame query result.
```

> > **Note:** For pre-v8 backward-compatibility there's an exception to the above rule when a query returns 3 columns including a string column named metric. Instead of transforming the metric column into field labels, it becomes the field name, which makes the series name be formatted as the value of the metric column. See example with metric column below.

Reviewer: ```suggestion
> For backward compatibility, there's an exception to the above rule for queries that return three columns including a string column named metric. Instead of transforming the metric column into field labels, it becomes the field name, and then the series name is formatted as the value of the metric column. See the example with the metric column below.
```

You can optionally customize the default series name formatting using instructions in [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}).

**Example with `metric` column:**

```sql
SELECT
  $__timeGroup(time_date_time, '5m') as time,
  min("value_double"),
  'min' as metric
FROM test_data
WHERE $__timeFilter(time_date_time)
GROUP BY $__timeGroup(time_date_time, '5m')
ORDER BY 1
```

Data frame result:

```text
+---------------------+-----------------+
| Name: time          | Name: min       |
| Labels:             | Labels:         |
| Type: []time.Time   | Type: []float64 |
+---------------------+-----------------+
| 2020-01-02 03:05:00 | 3               |
| 2020-01-02 03:10:00 | 6               |
+---------------------+-----------------+
```

**Example using the fill parameter in the $\_\_timeGroup macro to convert null values to be zero instead:**

```sql
SELECT
  $__timeGroup(createdAt, '5m', 0) as time,
  sum(value) as value,
  hostname
FROM test_data
WHERE
  $__timeFilter(createdAt)
GROUP BY
  $__timeGroup(createdAt, '5m', 0),
  hostname
ORDER BY 1
```

> Given the data frame result below and using the graph panel you'll get 2 series named _value 10.0.1.1_ and _value 10.0.1.2_ given the name of the field and its labels. If you would like to render the series with a name of _10.0.1.1_ and _10.0.1.2_ you could use a [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}) value of `${__field.labels.hostname}`.

Reviewer: ```suggestion
Given the data frame result in the following example and using the graph panel, you will get two series named _value 10.0.1.1_ and _value 10.0.1.2_ given the name of the field and its labels. To render the series with a name of _10.0.1.1_ and _10.0.1.2_ , use a [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}) value of `${__field.labels.hostname}`.
```

Reviewer: Can we avoid using "given" twice in the first sentence, that will make it an easier read.

Author: ```suggestion
Given the data frame result in the following example and using the graph panel, you will get two series named _value 10.0.1.1_ and _value 10.0.1.2_. To render the series with a name of _10.0.1.1_ and _10.0.1.2_ , use a [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}) value of `${__field.labels.hostname}`.
```

## Time series queries

> If you set Format as to Time series, then the query must have a column named time that returns either a SQL datetime or any numeric datatype representing Unix epoch in seconds. You can additionally return a column named metric that represents the metric type for the value column, for example, seconds. Any column except time and metric is treated as a value column. If you omit the metric column, then the title of the value column is treated as its metric type. When you select multiple value columns without a column named metric, then each column's title is treated as its metric type. 

Reviewer 2: If it works like this, I don't see why it should require the column to named time. I imagine it could be "a numeric type unix timestamp named time, or a datetime type returned from the SQL query.

Author: Currently, due to legacy reasons :smile:

Author: Definitely room for improvements in the future.

> If you set `Format as` to `Time series`, for use in Graph panel for example, then the query must have a column named `time` that returns either a SQL datetime or any numeric datatype representing Unix epoch in seconds. You may return a column named `metric` that is used as metric name for the value column. Any column except `time` and `metric` is treated as a value column. If you omit the `metric` column, the name of the value column will be the metric name. You may select multiple value columns, each will have its name as metric.
> If you return multiple value columns and a column named `metric` then this column is used as prefix for the series name (only available in Grafana 5.3+).
> 
> If you're not satisfied with the default series name formatting, please use the [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}) to customize it.

Reviewer: ```suggestion
If you set Format as to Time series, then the query must have a column named time that returns either a SQL datetime or any numeric datatype representing Unix epoch in seconds. You can additionally return a column named metric that represents the metric type for the value column, for example, seconds. Any column except time and metric is treated as a value column. If you omit the metric column, then the title of the value column is treated as its metric type. When you select multiple value columns without a column named metric, then each column's title is treated as its metric type. 

If you return multiple value columns and a column named metric, then the metric column is used as a prefix for the series name. You can optionally customize the default series name formatting using instructions in [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}).
```
